### PR TITLE
`vms/platformvm`: Permit usage of the `Transactions` field in `BanffProposalBlock`

### PR DESCRIPTION
**Pull request by dhrubabasu**
## Why this should be merged

This is an alternative approach to https://github.com/ava-labs/avalanchego/pull/2439 and https://github.com/ava-labs/avalanchego/pull/2442

The key difference is we apply the standard tx changes to a new `onDecisionState`. This gets applied first before the `onCommitState` and `onAbortState`. This allows us to leave the memory cadences untouched and only modify the `verifier` logic.

## How this works

* process the standard txs into an ephemeral `onAcceptState`
* Apply `onAcceptState` to both `onCommitState` and `onAbortState`
* Modify the proposal block `blockState` to include `standardBlockState`s (https://github.com/ava-labs/avalanchego/pull/2450)

## How this was tested

- [x] CI
- [x] Added UT with a proposal block containing a decision tx (`AddValidatorTx`)

### CODE REVIEW

**Comment by dhrubabasu** [comment]
We now only apply the `onDecisionState` once instead of twice

File: vms/platformvm/block/executor/verifier.go [hunk 1]
```diff
@@ -50,7 +50,7 @@ func (v *verifier) BanffCommitBlock(b *block.BanffCommitBlock) error {
 }
 
 func (v *verifier) BanffProposalBlock(b *block.BanffProposalBlock) error {
-	if len(b.Transactions) != 0 {
```
Comment (patrick-ogrady) [inline vms/platformvm/block/executor/verifier.go:53]: This was certainly forward thinking of us.

File: vms/platformvm/block/executor/verifier.go [hunk 2]
```diff
@@ -85,7 +85,18 @@ func (v *verifier) BanffProposalBlock(b *block.BanffProposalBlock) error {
 	onAbortState.SetTimestamp(nextChainTime)
 	changes.Apply(onAbortState)
 
-	return v.proposalBlock(&b.ApricotProposalBlock, onCommitState, onAbortState)
+	// Apply the changes, if any, from processing the decision txs.
+	onDecisionState, err := wrapState(onCommitState)
```
Comment (patrick-ogrady) [inline vms/platformvm/block/executor/verifier.go:89]: Does it matter that `onDecisionState` wraps `onCommitState` when applying in `abort`?

Comment (dhrubabasu) [reply vms/platformvm/block/executor/verifier.go:89]: Nope, at this point onCommitState and onAbortState are the same.

Comment (patrick-ogrady) [reply vms/platformvm/block/executor/verifier.go:89]: Good thing comments are free and you have room to elaborate on this 👀

Comment (dhrubabasu) [reply vms/platformvm/block/executor/verifier.go:90]: added a comment

File: vms/platformvm/block/executor/verifier.go [hunk 3]
```diff
@@ -86,6 +86,7 @@ func (v *verifier) BanffProposalBlock(b *block.BanffProposalBlock) error {
 	changes.Apply(onAbortState)
 
 	// Apply the changes, if any, from processing the decision txs.
+	// [onCommitState] = [onAbortState] here, either one can be used.
```
Comment (patrick-ogrady) [inline vms/platformvm/block/executor/verifier.go:89]: nit: `[onCommitState] == [onAbortState] here, so either one can be used to construct pending state changes. We apply these changes [TODO: where?] to whichever state is being committed, so the reference to state specified here is not important.`

**Review by patrick-ogrady — APPROVED** [review]

**Review by StephenButtolph — APPROVED** [review]

**Pull request merged**